lib/galaxy/model/migrations/util.py

```
@@ -1,6 +1,38 @@
+import logging
+from contextlib import contextmanager
+
 from alembic import op
+from sqlalchemy.exc import (
+    OperationalError,
+    ProgrammingError,
+)
+
+log = logging.getLogger(__name__)
 
 
 def drop_column(table, column):
     with op.batch_alter_table(table) as batch_op:
         batch_op.drop_column(column)
+
+
+@contextmanager
+def ignore_add_column_error(table, column):
```

We've always been ignoring this unconditionally in the past, this looks like a good idea.

---

@mvdbeek Yes, and I think that was a little risky: we silenced all exceptions unconditionally. However if that really helps in development (or in some non-dev edge case scenarios), maybe a safer option would be to have a config option? Then, if that option is set, all *relevant* errors are silenced and logged (but not a blanket `except Exception` like we had), but if not - the error is raised. It'll make it easier to write revision scripts too. The config option could be `ignore_migration_errors` (it does add yet another config option to our schema, but I don't think combining it with the `debug` option is the right thing to do). 
What do you think?

---

Case by case as you've done here is the right thing, don't think that warrants an extra option.